readthedocs/organizations/tests/test_filters.py

```diff
@@ -0,0 +1,624 @@
+from django.test.client import RequestFactory
+from pytest_django.asserts import assertQuerySetEqual
+
+from readthedocs.organizations.filters import (
+    OrganizationListFilterSet,
+    OrganizationProjectListFilterSet,
+    OrganizationTeamListFilterSet,
+    OrganizationTeamMemberListFilterSet,
+)
+from readthedocs.organizations.models import Organization, Team
+from readthedocs.projects.models import Project
+
+
+@pytest.mark.django_db
+class OrganizationFilterTestCase:
+    @pytest.fixture(autouse=True)
+    def set_up(self, settings):
+        settings.RTD_ALLOW_ORGANIZATIONS = True
+
+        self.owner = fixture.get(User)
+        self.user = fixture.get(User)
+        self.project = fixture.get(Project)
+
+        self.organization = fixture.get(
+            Organization,
+            owners=[self.owner],
+            projects=[self.project],
+        )
+        self.team = fixture.get(
+            Team,
+            access="admin",
+            organization=self.organization,
+            members=[self.user],
+            projects=[self.project],
+        )
+        self.team_empty = fixture.get(
+            Team,
+            access="readonly",
+            organization=self.organization,
+            members=[],
+            projects=[],
+        )
+
+        # Create extra objects just to ensure there are no extra objects in our
+        # filter querysets
+        self.other_owner = fixture.get(User)
+        self.other_user = fixture.get(User)
+        self.other_project = fixture.get(Project)
+        self.other_organization = fixture.get(
+            Organization,
+            owners=[self.other_owner],
+            projects=[self.other_project],
+        )
+        self.other_team = fixture.get(
+            Team,
+            access="admin",
+            organization=self.other_organization,
+            members=[self.other_user],
+            projects=[self.other_project],
+        )
+
+        self.set_up_extra()
+
+    def set_up_extra(self):
+        pass
+
+
+class TestOrganizationFilterSet(OrganizationFilterTestCase):
+    def get_filterset_for_user(self, user, *args, **kwargs):
+        request = RequestFactory().get("")
+        request.user = user
+        queryset = Organization.objects.for_user(user)
+        kwargs.update(
+            {
+                "queryset": queryset,
+                "request": request,
+            }
+        )
+        return OrganizationListFilterSet(*args, **kwargs)
+
+    @pytest.mark.parametrize(
+        "get_params",
+        [
+            lambda self: (self.user, self.organization),
+            lambda self: (self.owner, self.organization),
+            lambda self: (self.other_user, self.other_organization),
+            lambda self: (self.other_owner, self.other_organization),
+        ],
+    )
+    def test_unfiltered_queryset(self, get_params):
+        """No active filters returns full queryset."""
+        (user, organization) = get_params(self)
```

Comment:
@stsewd Parametrized lambdas work for this use case, but it's a little weird. I wasn't sure if there was a better way to use pytest `fixture()` to accomplish this. Is there a better pattern here?

Reply:
Nevermind, I used `pytest.fixture` in a more standard way. It works much nicer!